test/evp_test.c

```diff
@@ -3029,6 +3048,14 @@ static int parse(EVP_TEST *t)
     }
 
     for (pp++, i = 1; i < t->s.numpairs; pp++, i++) {
+        if (strcmp(pp->key, "Availablein") == 0) {
+            if (!prov_available(pp->value)) {
```

Comment:
Should we skip or run and expect failure?

Reply:
Skip, like the next few lines indicate... Or do I misunderstand the question?

Reply:
I am asking if instead of doing the skipping... It could run and expect to fail..

We at least need some negative testing somewhere, that tests that algorithms are not available when they shouldnt be...

Reply:
Hmmm, so something conditional. That would be a conditional "Reason" the... Trouble is that evp_test isn't at all designed for such things, at least now. It's an idea, sure, but unfitting for this PR.